### PR TITLE
Replacing: slf4j with jul

### PR DESCRIPTION
**Pull request by Ant-hem**
Replaced SLF4J with Java.Util.logging for tracing requests/responses

Removing slf4j allow to remove on more dependency of the library and
to simplify the shading process.

 - Added: JUL in HttpTransport with Log.FINEST level
 - Removed: SLF4J and Logback from pom.xml
 - Updated: logging.properties

 To activate JUL please set in logging.properties
 .level=FINEST
 com.algolia.search.HttpTransport=FINEST
 java.util.logging.ConsoleHandler.level=FINEST

 and set:
 -Djava.util.logging.config.file="logging.properties-path"

### CODE REVIEW

File: algoliasearch-core/src/main/java/com/algolia/search/HttpTransport.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
   private final HttpRequester httpRequester;
   private final RetryStrategy retryStrategy;
   private final ConfigBase config;
-  private static final Logger logger = LoggerFactory.getLogger("algoliasearch");
+  private static final Logger LOGGER = Logger.getLogger(HttpTransport.class.getName());
```
Comment (BenoitPerrot) [inline algoliasearch-core/src/main/java/com/algolia/search/HttpTransport.java:30]: For another commit: ensure all static-final are uppercase:
- make javaVersion uppercase in Config.base
- if possible add a rule in the formatting configuration

File: algoliasearch-core/src/test/resources/logging.properties [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
+# Even though the root logger has the same level as this,
+# the next line is still needed because we're configuring a handler,
+# not a logger, and handlers don't inherit properties from the root logger.
 java.util.logging.ConsoleHandler.level=INFO
+
+# Specify logging levels for specific namespaces.
+com.algolia.search.HttpTransport=INFO
 org.junit.platform.launcher.core.LauncherConfigurationParameters.level=WARNING
 org.junit.jupiter.engine.config.EnumConfigurationParameterConverter.level=WARNING
+org.apache.http.level=WARNING
```
Comment (BenoitPerrot) [inline algoliasearch-core/src/test/resources/logging.properties]: -1: Add back the carriage return at the end of the line

File: pom.xml [hunk 3]
```diff
@@ -208,11 +201,6 @@
                             <configuration>
                                 <shadedClassifierName>uber</shadedClassifierName>
                                 <promoteTransitiveDependencies>true</promoteTransitiveDependencies>
```
Comment (BenoitPerrot) [inline pom.xml]: -1: Remove this promoteTransitiveDependencies, as now unnecessary

This was introduced to have "algoliasearch-apache-uber" list "slf4j-api" as a dependency in its reduced pom.

**Review by BenoitPerrot — CHANGES_REQUESTED** [review]

**Comment by Ant-hem** [comment]
All good @BenoitPerrot ! I will submit another PR with the formatting rules later :)

File: algoliasearch-core/src/test/resources/logging.properties (hunk 2 above)
Comment (BenoitPerrot) [inline algoliasearch-core/src/test/resources/logging.properties]: Hmm 🤔 This shouldn't be necessary

Comment (Ant-hem) [reply algoliasearch-core/src/test/resources/logging.properties]: Good catch 😄

**Review by BenoitPerrot — APPROVED** [review]

**Pull request merged**